pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -190,6 +193,45 @@ const (
 	KernelTunableModify KernelTunableBehavior = "modify"
 )
 
+// Init creates the top level qos cgroup containers
+// We currently create containers for the Burstable and Best Effort containers
+// All guaranteed pods are nested under the RootContainer by default
+// Init is called only once during kubelet bootstrapping.
+func InitQOS(qosConfig *QOSConfig) (QOSContainersInfo, error) {
```

Comment:
nit: If you accept pointer arguments, you should generally validate that it is not nil.